FXyz-Importers/src/main/java/org/fxyz3d/importers/maya/MayaImporter.java

```diff
@@ -32,134 +32,103 @@
+        // It needs to be added by the user of MayaImporter.
+        //            root = new Xform();
+
+        MayaGroup mayaRoot = new MayaGroup();
+        // Add top level nodes to the root
+        int nodeCount = 0;
+        for (Node n : loader.loaded.values()) {
+            if (n != null) {
+                // Only add a node if it has no parents, ie. top level node
+                if (n.getParent() == null) {
+                    if (Loader.DEBUG) {
+                        System.out.println("Adding top level node " + n.getId() + " to root!");
+                    }
+                    n.setDepthTest(DepthTest.ENABLE);
+                    if (!(n instanceof MeshView) || ((TriangleMesh)((MeshView)n).getMesh()).getPoints().size() > 0) {
+                        mayaRoot.getChildren().add(n);
                     }
-                    nodeCount++;
                 }
+                nodeCount++;
             }
-            // rootCharacter.setRootJoint(loader.rootJoint);
-            if (Loader.DEBUG) { System.out.println("There are " + nodeCount + " nodes."); }
-
-            // if meshes were not loaded in the code above
-            // (which they now are) one would need to
-            // set meshParents from the loader
-            // meshParents.addAll(loader.meshParents.keySet());
-            // this is not necessary at the moment
-
-            timeline = new Timeline();
-            int count = 0;
-
-            // Add all the keyframes to the timeline from loader.keyFrameMap
-            for (final Map.Entry<Float, List<KeyValue>> e : loader.keyFrameMap.entrySet()) {
-                // if (DEBUG) System.out.println("key frame at : "+ e.getKey());
-                timeline.getKeyFrames().add
-                        (
-                                new KeyFrame(
-                                        javafx.util.Duration.millis(e.getKey() * 1000f),
-                                        (KeyValue[]) e.getValue().toArray(new KeyValue[e.getValue().size()])));
-                count++;
-            }
+        }
+        // rootCharacter.setRootJoint(loader.rootJoint);
+        if (Loader.DEBUG) {
+            System.out.println("There are " + nodeCount + " nodes.");
+        }
+
+        // if meshes were not loaded in the code above
+        // (which they now are) one would need to
+        // set meshParents from the loader
+        // meshParents.addAll(loader.meshParents.keySet());
+        // this is not necessary at the moment
+
+        Timeline timeline = new Timeline();
 
-            if (Loader.DEBUG) { System.out.println("Loaded " + count + " key frames."); }
-        } catch (Exception e) {
-            throw new RuntimeException(e);
+        loader.keyFrameMap.forEach((seconds, keyValues) -> {
+            timeline.getKeyFrames().add(
+                    new KeyFrame(Duration.millis(seconds * 1000), keyValues.toArray(new KeyValue[0]))
+            );
+        });
+
```

Comment:
I would collect the `KeyFrame`s with:

    loader.keyFrameMap.entrySet().stream()
            .map(entry -> new KeyFrame(Duration.seconds(entry.getKey()), null, null, entry.getValue()))
            .forEach(timeline.getKeyFrames()::add);

* Stream allows us the future option of doing it in parallel, which, considering the amount of animation frames, seems relevant.
* `Duration.seconds` replaces `Duration.millis` and then multiplying by 1000.
* Using the `KeyFrame` constructor that takes a collection instead of an array saves the copying of `toArray`, which can save a lot of garbage. The 2nd and 3rd `null` arguments are the default. For more info see [JDK-8091817](https://bugs.openjdk.java.net/browse/JDK-8091817).

Reply:
Great alternative.